bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -92,6 +105,21 @@ protected void deactivate() {
         pageChangeListeners.clear();
     }
 
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        applyConfig(config);
+    }
+
+    private void applyConfig(Map<String, Object> config) {
+        if (config == null) {
+            return;
+        }
+        final String max = Objects.toString(config.get("maxSubscriptions"), null);
+        if (max != null) {
+            maxSubscriptions = Integer.parseInt(max);
```

Comment:
Shouldn't you handle the NumberFormatException?

Reply:
Yes, not a bad idea.